### PR TITLE
Ensure Fleet works on k8s 1.22 and bump tekton-utils/gitjob

### PR DESCRIPTION
**Pull request by nickgerace**
## Relevant issues
- https://github.com/rancher/rancher/issues/33586
- https://github.com/rancher/rancher/issues/33112
- https://github.com/rancher/fleet/issues/451

## Affected APIs:
```
admissionregistration.k8s.io/v1beta1 API version of MutatingWebhookConfiguration and ValidatingWebhookConfiguration
apiextensions.k8s.io/v1beta1 API version of CustomResourceDefinition
apiregistration.k8s.io/v1beta1 API version of APIService
authentication.k8s.io/v1beta1 API version of TokenReview
authorization.k8s.io/v1beta1 API version of LocalSubjectAccessReview, SelfSubjectAccessReview, and SubjectAccessReview
certificates.k8s.io/v1beta1 API version of CertificateSigningRequest
coordination.k8s.io/v1beta1 API version of Lease
extensions/v1beta1 API versions of Ingress
networking.k8s.io/v1beta1 API versions of Ingress and IngressClass
rbac.authorization.k8s.io/v1beta1 API version of ClusterRole, ClusterRoleBinding, Role, and RoleBinding
scheduling.k8s.io/v1beta1 API version of PriorityClass
storage.k8s.io/v1beta1 API version of CSIDriver, CSINode, StorageClass, and VolumeAttachment
```

### CODE REVIEW

**Comment by brendarearden** [comment]
This ensures that there are no deprecation warnings for 1.22, but do these apis work on all rancher supported versions of k8s for the 2.6 release? Just want to make sure backwards compatibility is still okay.

**Comment by nickgerace** [comment]
@brendarearden: Yes! The only major API change that was a full switchover rather than a migration was [GitJob CRD](https://github.com/rancher/gitjob/commit/d463c2161974430b4b77db6ec580ea2ba8c0cfe3). It was added in [k8s v1.16](https://kubernetes.io/docs/reference/using-api/deprecation-guide/), which is the lowest SemVer of k8s we allow for [Rancher v2.5.0](https://rancher.com/support-maintenance-terms/all-supported-versions/rancher-v2.5.0) (which is when Fleet was released).

**Review by brendarearden — DISMISSED** [review]

**Comment by nickgerace** [comment]
This is on hold until this is resolved: https://github.com/rancher/rancher/issues/33586

**Comment by nickgerace** [comment]
Awaiting: https://github.com/rancher/wrangler/pull/171

**Comment by nickgerace** [comment]
Ready for re-review since https://github.com/rancher/rancher/issues/33586 was resolved.

File: charts/fleet-crd/templates/gitjobs-crds.yaml [hunk 1]
```diff
@@ -1,3 +1,3441 @@
+{{- if .Capabilities.APIVersions.Has "apiextensions.k8s.io/v1" -}}
```
Comment (StrongMonkey) [inline charts/fleet-crd/templates/gitjobs-crds.yaml:1]: How did you generate this? Did you also update gitjob CI to include your change?

File: charts/fleet/charts/gitjob/Chart.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 apiVersion: v2
-appVersion: 0.1.15
-description: Controller that run jobs based on git events
 name: gitjob
-version: 0.1.15
+description: Controller that run jobs based on git events
```
Comment (StrongMonkey) [inline charts/fleet/charts/gitjob/Chart.yaml:3]: The latest release is 0.1.17(@jiaqiluo Added cve fix)

Comment (StrongMonkey) [reply charts/fleet/charts/gitjob/Chart.yaml:3]: Also you might need to copy the chart over. There are other changes besides image and version bump.

File: charts/fleet-crd/templates/gitjobs-crds.yaml (hunk 1 above)
Comment (nickgerace) [reply charts/fleet-crd/templates/gitjobs-crds.yaml:1]: Yes, the GitJob PR covers it: https://github.com/rancher/gitjob/pull/36

Oops, I forgot to add you for review there.

File: charts/fleet/charts/gitjob/Chart.yaml (hunk 2 above)
Comment (nickgerace) [reply charts/fleet/charts/gitjob/Chart.yaml:3]: I think we should look at the GitJob PR first, actually: https://github.com/rancher/fleet/pull/441#discussion_r677794034

Comment (nickgerace) [reply charts/fleet/charts/gitjob/Chart.yaml:3]: Fixed

File: charts/fleet-crd/templates/gitjobs-crds.yaml (hunk 1 above)
Comment (nickgerace) [reply charts/fleet-crd/templates/gitjobs-crds.yaml:1]: Generated with the GitJob CRD gen.
```sh
( cd rancher-gitjob; go run pkg/crdgen/main.go > ../rancher-fleet/charts/fleet-crd/templates/gitjobs-crds.yaml )
```

**Review by StrongMonkey — APPROVED** [review]

**Pull request merged**